### DISCUSSION TITLE
Feature Request: Time/Date Range for a scenario

### DISCUSSION THREAD

**Discussion by Race666** [MMRIZE/MMM-Scenes2#1] [Ideas]
Some scenarios, like the school time table for a child , I just need at a specfic weekday and only in the morning. Would be possible to add two options to scenario at which weekday and which time a role is only being entered? Something like that this example:

    enter: [
      {
        role: "role1",
        animation: "bounceIn",
        duration: 1500,
        gap: 100,
        // Mon-Fr
        weekday: [0,1,2,3,4,5],
        // A given time range 6:00 - 7:59
        timeRanges: [ "6-7" ]
        // or 6:00 - 8:59
        timeRanges: [ 6,7,8 ]
      },

**Comment by eouia** [comment]
Interesting. I’ll consider it

**Comment by eouia** [comment]
Could you explain exactly what you need?
- Special `scene` for the specific period (e.g. `Scene1 - Scene 2 - Scene3` for working, `Scene4 - Scene2 - Scene3` for holidays)
- Special `role` in the specific scene for the specific period (e.g. `role1, role2 in Scene1` for working, `role1, role3 in Scene1` for holidays)
- Or reconfigurable the whole scenario on the fly.
>   All results would be similar to your needs. But different approach.

**Reply by Race666** [reply]
For my needs something link this would be great:

In your name: A Special Scence:

On specific weekdays (for example weekdays: [0,1,2,3,4,5] ) and at specfic time range (for example 6:00am - 9:00am) :

Scene1 - Scene 2 - Scene3

otherwise 

Scene1 - Scene 2 

Means: An options/properties on which weekdays and time range(s)  a scene is active.

In a further step a property for holidays (may be by query the  calender modul with for example (https://www.schulferien-online.de/wp-content/uploads/2023/10/schulferien_by-24.ics) would also be cool :-)

**Comment by eouia** [comment]
I implemented it, but it's a bit different from the configuration method you wanted. I could have done it the way you suggested, but I wanted a more universally expandable form. That's because, as I thought about it, there would be many desires to control the flow of the scenario with various conditions and forms other than just time.
So instead of making it detailed with all the imaginable option items, I made it so that anyone can control it more freely with a callback function. However, it might be difficult for those who don't know much about programming.
I posted a sample of the form you wanted as an example on the wiki page. I hope it helps, and if you have any difficulties, please ask again.
https://github.com/MMRIZE/MMM-Scenes2/wiki/Dynamic-branching-by-callback-function

**Reply by eouia** [reply]
More to read.

- https://github.com/MMRIZE/MMM-Scenes2?tab=readme-ov-file#previousnext-in-scene-since-110
- https://github.com/MMRIZE/MMM-Scenes2/wiki

**Comment by Race666** [comment]
Thank you! Will take a look on it in the next days.

**Comment by Race666** [comment]
Great. Works  :-) Thank you
